rules/framework/vfs_overlay.bzl

```diff
@@ -232,9 +232,11 @@ def _framework_vfs_overlay_impl(ctx):
 
     # Conditionally collect and pass in the VFS overlay here.
     virtualize_frameworks = feature_names.virtualize_frameworks in ctx.features
+    additional_cc_infos = []
     if virtualize_frameworks:
         for dep in ctx.attr.deps:
             if FrameworkInfo in dep:
+                additional_cc_infos.append(dep[CcInfo])
```

Comment:
@maxwellE Does this end up getting it working for you? I think one of the main issues is that isn't setting all files / swiftmodules in the returned `FrameworkInfo` provider within `apple_patched.bzl`. That rule could ideally declaratively specify the framework info 

Take this with a grain of salt, but it _may_ work by getting the info from  the `legacy_target`'s `AppleImportedFrameworkInfo` in `_get_framework_info_providers` e.g. the swiftmodules, and hand them over to `_build_subtrees` in the VFS rule. I remember there were edge case and would have to look into this a bit more.

Reply:
This is not fully working yet, I am trying to understand what is going on here, forgive me. 

Thank you for the tip, Id love to get this working and learn more about the inner workings

Reply:
Do you mean `AppleFrameworkImportInfo`? Not directly seeing how we can determine swiftmodules from that

Reply:
You've got it! Yeah, if you don't have a `SwiftInfo` provider, you possibly synthesize one from the data in https://github.com/bazelbuild/rules_apple/blob/master/doc/providers.md#appleframeworkimportinfo and do something similar to how `rules_apple` processes `framework_imports` 🤔

Reply:
Also if that - provider isn't the right one, you can try to get the swift modules from some other way. They should definitely be in the build graph for hermeticity